ios-target-app/Managers/LocationManager.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  LocationManager.swift
+//  ios-target-app
+//
+//  Created by Raul Piñeres Carrera on 13/10/22.
+//
+
+import UIKit
+import CoreLocation
+
+class LocationManager: NSObject, CLLocationManagerDelegate {
+  
+  static let instance = LocationManager()
+  private var locationManager: CLLocationManager = CLLocationManager()
+  
+  @Published var lastLocation: CLLocation?
+  
+  override init() {
+    super.init()
+    locationManager.delegate = self
+    locationManager.desiredAccuracy = kCLLocationAccuracyBest
+    
+    if CLLocationManager.locationServicesEnabled() { // Check for Location Services
+      locationManager.requestWhenInUseAuthorization()
+      locationManager.startUpdatingLocation()
```

Comment:
this IMO should be done in a public function that you expose, and consumers of LocationManager decide when to start/stop listening for location updates.
On a different topic, this should also check for the user permissions, I think `locationServicesEnabled` doesn't take that into account.

Reply:
mmm Ok this one only checks if it is enabled or not, right? but it is not that accurate.